### ISSUE TITLE
Set up repo

### ISSUE THREAD

**Issue by rrabinovitch** [rrabinovitch/Crate#3]
Instructions [here](https://mod4.turing.io/projects/crate/crate_setup.html)

When you're done with setup, answer the following questions in a comment below:
- what went well as you set up the application?
- what strategies did you use when running into errors?
- what things did you find helpful in the documentation?
- what notes do you want to make sure are included in the readme?

**Check the box next to your name when you've completed this item:**
- [x] Ruthie
- [x] Nick
- [x] Horacio
- [x] Taryn
- [x] Ash

**Comment by NickEdwin** [comment]
1) Setup went fairly smooth, hit a snag with my xcode(I believe it was) CLI interface not installing. As always some googling and some stack overflow links helped me to figure it out! The help channel now has double the people with double the issues but also double the help which is a nice resource too!
2) In the documentation I really like the "core setup" breakdown of the file tree, it helped to navigate that initially overwhelming tree and lock down where we needed to look. For my future README's I would copy from this: a) Detailed setup b) File tree or visual schema c) The resources and inspirations section

**Comment by H-Bo214** [comment]
1. The initial part of the set up went smoothly. I ran into issues on step 3 of the Postgres set up. The directions on the Turing site “Configuring you $PATH variable’’ has a code snippet as a string. This wasn’t working for me. I ended up reaching out in our team slack and received help from Nick to get past this. Additionally, I was missing a semicolon when trying to create the Postgres Database. I received help from Ruthie on this issue. I ran into another issue that had already been discussed in the cohort channel. The discussion helped solve this issue. 

2. The set up instructions in the README are really descriptive and specific.
* Have a very detailed README
* Hire Me section
* Resources and inspiration section

**Comment by rrabinovitch** [comment]
> what went well as you set up the application?

- The setup instructions provided by Turing and the original repo owner were mostly pretty clear, so aside from not knowing to change the dialect in the db json file things went relatively smoothly!

> what strategies did you use when running into errors?

- The primary error that got me stuck was when running `npm run setup` from within the `api` directory, which produced these error messages:

<img width="1026" alt="Screen Shot 2020-10-05 at 3 38 27 PM" src="https://user-images.githubusercontent.com/62635544/95395316-92384580-08bb-11eb-9da4-59c7d5e8ae2e.png">
<img width="1022" alt="Screen Shot 2020-10-05 at 3 33 06 PM" src="https://user-images.githubusercontent.com/62635544/95395310-8f3d5500-08bb-11eb-9a05-36884bc8a58f.png">
I've gotten so used to Ruby and Rails error messages, which read a lot more easily to me so it took some fumbling to figure out even what to google - but I eventually started by looking into the `ERROR: connect ECONNREFUSED 127.0.0.1:3306` message. I also explored the error log and looked into the `ELIFECYCLE` error that was appearing in there. Both approaches were primarily returning results that suggested running commands involving `force` or `sudo` which we've been told to be cautious of. So at that point I reached out to peers and eventually instructors, to realize that the `dialect` in the `/api/src/config/database.json` file needed to be changed to `postgres`. ([Slack thread](https://turingschool.slack.com/archives/C013KTYM86S/p1601934182023500) for reference)

This brought me to a different error, but it was a lot quicker to resolve via some googling.

> what things did you find helpful in the documentation?  

- I don't love needing to hop back and forth between multiple browser windows for set up instructions, but aside from that the setup instruction language and steps (from both Turing and the original repo owner) made it easy to follow. 

> what notes do you want to make sure are included in the readme?  

- We should make sure to include notes on [Turing's setup instructions](https://mod4.turing.io/projects/crate/crate_setup.html) as well as a note on updating the database dialect in the `database.json` file

**Comment by tarynmartin** [comment]
- went well: before and after postgres was easy enough
- issues: asked cohort for help/feedback on what I was doing wrong. Also looked at other conversations to see what others had done with similar/same issues
- documentation: file tree structure and setup instructions
- include turing set up instructions w/maybe clarification on how to get to .zshrc (first time I've used it in months and was having trouble with it); include setup & file tree; maybe keep some screen shots, but think we should make gifs to put up instead.

**Comment by Ashkanthegreat** [comment]
- Ran into issues with `npm run setup` because Postgres wasn't setup properly. 
- Also, for some reason there was "a server already running" on my computer and would not turn off, even if I restarted my machine.  Had to do some research on how to force it to turn off.
- I also did not like jumping from multiple pages to setup. I think it can be combined and updated for .zshrc, since no student from now on will be using .bash. 
- For future README's, I would like to see a more detailed setup instructions, so I wouldn't have to reference Turing's.
